source/auth/auth.md

```diff
@@ -1306,6 +1323,10 @@ in the MONGODB-OIDC specification, including sections or blocks that specificall
         check MUST be performed after SRV record resolution, if applicable. This property is only required for drivers
         that support the [Human Authentication Flow](#human-authentication-flow).
 
+    - AWS_CREDENTIAL_PROVIDER
+
+        A function or object from the AWS SDK that can be used to return AWS credentials.
```

Comment:
Consider similarly allowing the property to be defined on the `MongoClient` (as is done for `OIDC_HUMAN_CALLBACK`):


```suggestion
        A function or object from the AWS SDK that can be used to return AWS credentials. Drivers MAY allow the user to
        specify the callback using a `MongoClient` configuration instead of a mechanism property, depending on what is
        idiomatic for the driver.
```

I expect this may be preferred over specifying in the URI string.